Foundation/NSJSONSerialization.swift

```diff
@@ -322,6 +326,93 @@ private struct JSONWriter {
         }
     }
 
+    private func serializeUInt(value: UInt) throws {
+        if value == 0 {
+            writer("0")
+            return
+        }
+        var array: [UInt] = []
+        var stringResult = ""
+        //Maximum length of an UInt
+        array.reserveCapacity(20)
```

Comment:
I think we could make these constants like this:

```swift
let maxUIntLength = String(describing: UInt.max).characters.count
let maxIntLength = String(describing: Int.max).characters.count
```

We'd need to calculate these once (not every time we serialize), so make them private properties of `NSJSONSerialization`.